include/boost/geometry/extensions/gis/io/wkb/write_wkb.hpp

```diff
@@ -25,7 +25,7 @@ template <typename G>
 struct write_wkb<point_tag, G>
 {
     template <typename OutputIterator>
-    static inline bool write(const G& geometry, OutputIterator& iter, 
+    static inline bool write(const G& geometry, OutputIterator iter,
```

Comment:
It's because otherwise it wouldn't work on a standard-conformant compiler for a r-value type passed as the iterator. In the tests `std::back_insert_iterator` returned by `std::back_inserter()` is passed into this function which doesn't compile e.g. on GCC. This is why the iterators in the STL algorithms are always passed by value and if needed returned from the function.

In this particular case I guess the iterator could be passed by reference into the `dispatch::write_wkb()` but still it must be passed by value into the `geometry::write_wkb()`. I passed it by value also in the internals the same way how it's done in the STL.